hasher.go

```diff
@@ -33,26 +31,40 @@ type ValueHasher interface {
 	ValueHashSize() int
 }
 
+// trieHasher is a common hasher for all trie hashers (paths & values).
 type trieHasher struct {
 	hasher    hash.Hash
 	zeroValue []byte
 }
+
+// pathHasher is a hasher for trie paths.
 type pathHasher struct {
 	trieHasher
 }
+
+// valueHasher is a hasher for leaf values.
 type valueHasher struct {
 	trieHasher
 }
 
-func newTrieHasher(hasher hash.Hash) *trieHasher {
+type nilPathHasher struct {
+	hashSize int
+}
+
+// NewTrieHasher returns a new trie hasher with the given hash function.
+func NewTrieHasher(hasher hash.Hash) *trieHasher {
 	th := trieHasher{hasher: hasher}
 	th.zeroValue = make([]byte, th.hashSize())
 	return &th
 }
 
+func NewNilPathHasher(hasherSize int) PathHasher {
```

Comment:
I don't think this should be exported AFAIK the validation logic for the ClosestProof automatically handles this.